### DISCUSSION TITLE
What's wrong in my RepeatableEntry ->state nothing return in TextEntry?

### DISCUSSION THREAD

**Discussion by devhoussam1998** [filamentphp/filament#15179] [Help]
### Package

Infolist builder

### Package Version

v3.2.131

### How can we help you?

Here is what I get:

<img width="504" alt="Screenshot 2024-12-25 at 12 37 23" src="https://github.com/user-attachments/assets/40e57f7f-7368-4f39-aef6-35eb8965a41f" />

Current Code:

```
                    Action::make('urls')
                        ->infolist([
                            RepeatableEntry::make('urls')
                                ->state(function (Page $record) {
                                    $record->load('directories');

                                    $urls = collect($record->directories)
                                        ->map(function ($directory) use ($record) {
                                            $baseUrl = $directory->custom_domain ?? $directory->slug . '.' . config('app.short_url');

                                            return [
                                                'url' => "https://{$baseUrl}/{$record->slug}",
                                            ];
                                        })
                                        ->toArray();

                                    return $urls;
                                })
                                ->schema([
                                    TextEntry::make('url'),
                                ]),
                        ]),
```

**Comment by devhoussam1998** [comment]
still no comment?

**Reply by tahinuralam** [reply]
I know how to display only single row 
Not all rows

**Reply by devhoussam1998** [reply]
i want to display all rows

**Reply by devhoussam1998** [reply]
also this not work

```
RepeatableEntry::make('urls')
->state([
['url' => '#'],
['url' => '#'],
])
->schema([
TextEntry::make('url'),
])
```

**Reply by tahinuralam** [reply]
https://github.com/filamentphp/filament/discussions/13950

**Reply by tahinuralam** [reply]
> also this not work
> 
> ```
> RepeatableEntry::make('urls')
> ->state([
> ['url' => '#'],
> ['url' => '#'],
> ])
> ->schema([
> TextEntry::make('url'),
> ])
> ```

Use this 
https://github.com/filamentphp/filament/discussions/13950


Knock me on

https://wa.me/qr/SN4XAXSE7DYUO1

**Comment by jwhoami** [comment]
This is loosely related, but just wanted to document here, since the documentation does not mention it and I kinda stumbled upon it

`Infolists\Components\RepeatableEntry::make('children')`

The "children" can be a json field but it can also be a relation on your model and so you can repeat fields of the relation, in this case children.
